metrixpp/tests/general/test_std_code_member/sources/test.cpp

```diff
@@ -0,0 +1,59 @@
+#include "test.hpp"
+#include <iostream>
+
+// false positive: the parser cannot identify these 3 lines as a single statement
+   typedef typename boost::intrusive::
+      pointer_traits<void_pointer>::template
+         rebind_pointer<block_ctrl>::type                         block_ctrl_ptr;

```

Comment:
Here I hit a problem, can this be solved on the parser level somehow?

Reply:
As per my comment here: https://github.com/metrixplusplus/metrixplusplus/issues/42#issuecomment-735456563
Can we just count `;` symbols to get number of fields within a class? It is certainly best way of finding god classes.

> "Here I hit a problem, can this be solved on the parser level somehow?"
This is detected as a field. Why would not you want to count it as a member? It is type defining member. And I consider well contributes to spotting large classes.

I guess the goal of the tool was to:
- allow spotting of `bad quality code` using `good enough` counting with simple implementation and fast performance Over
- precise counting for everything and poor performance. 

So, to answer your question - I do not think there is anything within the parser to differentiate different types of fields.

What are the other "false positives" left? I just realized that this mentioned false positive was not considered a false positive in my code bases when I applied the tool back in 2008-2013 years. I wanted to make classes not big regardless of of what made them big...